src/components/cleaner.ts

```diff
@@ -70,4 +86,38 @@ export class Cleaner {
             })
         })
     }
+
+    private cleanCommand(rootFile: string): Promise<void> {
+        const configuration = vscode.workspace.getConfiguration('latex-workshop')
+        const command = configuration.get('latex.clean.command') as string
+        let args = configuration.get('latex.clean.args') as string[]
+        if (args) {
+            args = args.map(arg => arg.replace('%TEX%', rootFile))
+        }
+        this.extension.logger.addLogMessage(`Clean temporary files using: ${command}, ${args}`)
+        return new Promise((resolve, reject) => {
+            const proc = cs.spawn(command, args, {cwd: path.dirname(rootFile), detached: true})
+            let stderr = ''
+            proc.stderr.on('data', newStderr => {
+                stderr += newStderr
+            })
+            proc.on('error', err => {
+                this.extension.logger.addLogMessage(`Cannot run ${command}: ${err.message}, ${stderr}`)
+                if (err instanceof Error) {
+                    this.extension.logger.logError(err)
+                }
+                reject(err)
+            })
+            proc.on('exit', exitCode => {
+                if (exitCode === 0) {
+                    resolve()
+                } else {
+                    this.extension.logger.addLogMessage(`The clean command failed with exit code ${exitCode}`)
+                    this.extension.logger.addLogMessage(`Clean command stderr: ${stderr}`)
+                }
```

Comment:
We should call `resolve` here. Otherwise, the promise object will not be resolved.